### PR TITLE
AB#180302: Adding Custom word

### PR DESCRIPTION
**Pull request by MoizIqbal97**
This PR allows custom words to be added to the wordlist

**KeyChanges**

- AddCustomWord method for backend
- AddCustomWordModel
- addCustomWord Api

### CODE REVIEW

**Review by Jivanam — DISMISSED** [review]

File: app/Decsys/Controllers/WordlistsController.cs [hunk 1]
```diff
@@ -173,12 +170,40 @@ public async Task<IActionResult> DeleteRule(string wordlistId, int ruleIndex)
...
+        public async Task<IActionResult> AddCustomWord(string wordlistId, [FromBody] WordlistWord wordlistWord)
+        {
+            if (string.IsNullOrWhiteSpace(wordlistWord.Type) || string.IsNullOrWhiteSpace(wordlistWord.Word))
+            {
+                return BadRequest("Invalid type or word.");
+            }
+
+            wordlistWord.Type = wordlistWord.Type.ToLowerInvariant();
+
+            if (wordlistWord.Type != "noun" && wordlistWord.Type != "adjective")
```
Comment (dcl10) [inline app/Decsys/Controllers/WordlistsController.cs:195]: perhaps not in this PR but another one, you could make an `enum` for word types? Then you can reuse it everywhere and not compare magic string like here

File: app/Decsys/Services/WordlistService.cs [hunk 2]
```diff
@@ -50,6 +50,9 @@ public async Task DeleteRule(string wordlistId, int ruleIndex)
 
     public async Task<WordlistWord> SetExcludedBuiltins (string wordlistId, string type, string word)
     => await _wordlist.SetExcludedBuiltins(wordlistId, type, word);
+    
+    public async Task<WordlistWord> AddCustomWord(string ownerId,string wordlistId, string type, string word)
```
Comment (dcl10) [inline app/Decsys/Services/WordlistService.cs:54]:
```suggestion
    public async Task<WordlistWord> AddCustomWord(string ownerId, string wordlistId, string type, string word)
```

File: app/Decsys/Repositories/LiteDb/LiteDbWordlistRepository.cs [hunk 3]
```diff
@@ -54,7 +54,12 @@ public Task<WordlistWord> SetExcludedBuiltins(string wordlistId, string type, st
     {
         throw new NotImplementedException();
     }
-    
+
+    public Task<WordlistWord> AddCustomWord(string ownerId, string wordlistId, string type, string word)
+    {
+        throw new NotImplementedException();
```
Comment (blargly) [inline app/Decsys/Repositories/LiteDb/LiteDbWordlistRepository.cs:60]: Is this intended for now?

Comment (MoizIqbal97) [reply app/Decsys/Repositories/LiteDb/LiteDbWordlistRepository.cs:60]: yes, liteDb is for the desktop app which doesn't support wordlist implementations. We've opted to go with only MongoDb for features that require an ownerId ( liteDb doesn't have users set up )

**Review by dcl10 — DISMISSED** [review]

File: app/Decsys/Models/Wordlist/WordType.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+namespace Decsys.Models.Wordlist;
+
+public enum WordType
+{
+    noun,
+    adjective
```
Comment (dcl10) [inline app/Decsys/Models/Wordlist/WordType.cs:6]:
```suggestion
    Noun,
    Adjective
```
⛏️ This looks more in line with the C# style guide

Comment (MoizIqbal97) [reply app/Decsys/Models/Wordlist/WordType.cs:6]: To keep consistent with how the database saves "noun" and "adjectives" in other properties I've decided to go with this

File: app/Decsys/Repositories/LiteDb/LiteDbWordlistRepository.cs (hunk 3 above)
Comment (MoizIqbal97) [reply app/Decsys/Repositories/LiteDb/LiteDbWordlistRepository.cs:60]: Yes, wordlist does not work for liteDb which is used for the Desktop app. The Desktop app doesnt have users set up and any feature that relies on users such as wordlist only used MongoDb.

**Review by dcl10 — DISMISSED** [review]

**Review by dcl10 — APPROVED** [review]

**Review by blargly — APPROVED** [review]

**Pull request merged**